### ISSUE TITLE
Add MUR citations to the legal/citations endpoint 

### ISSUE THREAD

**Issue by tmpayton** [fecgov/openFEC#5996]
### What we’re after

 We should add MUR citations to the legal/citations endpoint. 

### Related ticket(s)

(Include the tickets that either came before, after, or are happening in tandem with this new ticket)

- [ ] https://github.com/fecgov/openFEC/issues/5979

### Completion criteria

(What does the end state look like - as long as this task(s) is done, this work is complete)

- [ ] MUR citations can be found in the citations endpoint

**Comment by patphongs** [comment]
### Overview

@tmpayton Let's rescope your ticket to just adding MURs to the citations endpoint for now. There's some adjustments we need to make for ADRs before we can implement it for the /legal/citations endpoint.

#### Citations autosuggest /legal/citations/regulation/ and /legal/citations/statute/ 
Add MUR citations WITHOUT subsections (the stuff in parentheses). Ex: 11 CFR §110.5. If a user wants to search citation subsections, they can go to the legal docs keyword search.

```
{
   "citation_text": "11 CFR §110.5",
   "citation_type": "regulation",
   "type": "citations"
},

{
"citation_text": "52 U.S.C. §30104",
"citation_type": "statute",
"type": "citations"
},
```



#### Legal search results response /legal/search/
We use `case_regulatory_citation` to filter for MUR citations already. We don't need to make any changes to this endpoint.

**Of note:** When I attempted to put in subsections into the `case_regulatory_citation` query filter for /legal/search/, it did not use it, it simply returned the exact match on the title, part, and section. So it really doesn't matter if subsections were included anyways? Ex: https://api.open.fec.gov/v1/legal/search?api_key=DEMO_KEY&type=murs&case_regulatory_citation=11+CFR+%C2%A7110.5(a)(b)

### Additional notes about citations
- For MURs, we want the citations tied to the final violations. This is already the case now, so no changes needed here.
- The citations that are part of the final violations are split based on respondent.

**Comment by tmpayton** [comment]
We will add in archived murs at a later time: [link](https://fecgov.slack.com/archives/C3X3K6EVA/p1729099827184039?thread_ts=1729091048.117229&cid=C3X3K6EVA) to slack conversation.
